src/main/java/org/matsim/contrib/gtfs/GtfsConverter.java

```diff
@@ -1,244 +1,400 @@
+        Route route = feed.routes.get(trip.route_id);
+        return includeRouteType.test(route.route_type) && includeAgency.test(route.agency_id);
+    }
+
+    private void convertStops() {
+        for (Stop stop : feed.stops.values()) {
+            if (!filterStops.test(stop))
+                continue;
+
+            TransitStopFacility t = this.ts.getFactory().createTransitStopFacility(Id.create(stop.stop_id, TransitStopFacility.class), transform.transform(new Coord(stop.stop_lon, stop.stop_lat)), false);
+            t.setName(stop.stop_name);
+            ts.addStopFacility(t);
+        }
+    }
+
+
+    private List<String> getActiveServiceIds(Map<String, Service> services) {
+        List<String> serviceIds = new ArrayList<>();
+        log.info("Used Date for active schedules: " + this.date.toString() + " (weekday: " + date.getDayOfWeek().toString() + "). If you want to choose another date, please specify it, before running the converter");
+        for (Service service : services.values()) {
+            if (service.activeOn(date)) {
+                serviceIds.add(service.service_id);
+            }
+        }
+        return serviceIds;
+    }
+
+
+    private void convertTrips(List<Trip> trips) {
+        int scheduleDepartures = 0;
+        int frequencyDepartures = 0;
+        for (Trip trip : trips) {
+            if (feed.getFrequencies(trip.trip_id).isEmpty()) {
+                if (feed.getOrderedStopTimesForTrip(trip.trip_id) == null || !feed.getOrderedStopTimesForTrip(trip.trip_id).iterator().hasNext()) {
+                    log.error("Found a trip with neither frequency nor ordered stop times. Will not add any Matsim TransitRoute/Departure for that trip. GTFS trip_id=" + trip.trip_id);
+                    continue;
+                }
+                StopTime firstStopTime = feed.getOrderedStopTimesForTrip(trip.trip_id).iterator().next();
+                Double departureTime = Time.parseTime(String.valueOf(firstStopTime.departure_time));
+                List<TransitRouteStop> stops = new ArrayList<>();
+                try {
+                    for (StopTime stopTime : feed.getInterpolatedStopTimesForTrip(trip.trip_id)) {
+                        Id<TransitStopFacility> stopId = Id.create(stopTime.stop_id, TransitStopFacility.class);
+                        TransitStopFacility stop = ts.getFacilities().get(stopId);
+
+                        // This stop was filtered and will be ignored
+                        if (stop == null)
+                            continue;
+
+                        double arrivalOffset;
+                        if (stopTime.arrival_time != Integer.MIN_VALUE) {
+                            arrivalOffset = Time.parseTime(String.valueOf(stopTime.arrival_time)) - departureTime;
+                        } else {
+                            // TODO: should use optional time
+                            arrivalOffset = Double.NEGATIVE_INFINITY;
```

Comment:
Probably yes, better OptionalTime, because it can be undefined, but currently we cannot create a TransitRouteStop with OptionalTime arrivalOffset, so it would need changes in matsim-libs.